.travis.yml

```diff
@@ -1,41 +1,160 @@
-os:
-  - linux
-  - osx
+#  2020-05-26: NOTE: Enabling experimental Travis feature of YAML check inside
+# Look into: Build #NUM -> Job #NUM.N -> View config -> Build config validation
+version: ~> 1.0
+
+# NOTE: Please, be aware that Travis YAML & docs & API are hard to make work properly. Travis configuration requires a lot of retries and some compromises. Tere are many ways that may look like that can be done in that way, but it would not work most of the time, or not work the way that you expect, need it. Travis config works only the certain particular ways. Some things look possible - but they are impossible in Travis. Current configuration is "the best way possible" that was found in ~100-150-200 retries, depending on what concider a retry.
```

Comment:
No!..

I've just put a WARNING there.

I assembled different Travis configurations several times over last ~6 years. And Travis almost didn't got any better.

For example:
```
env:
  jobs:
     - ....

#AND

jobs:
  ...
    - ...
      env: ...
```

Allow only certain options in themself, but there is no docs what options they accept, and what is the difference between `jobs:` and `jobs:` and `env:` and `env:`, so in docs there is constant disambiguation of what is what and what doc talks about.

Documentation still does not have some very needed explanations and examples.

Travis and `shell` are very fragile until you got them right. Even prettifying the commit history caused major code destruction, and a long rebase and recheck process.

I've just put a warning that everything is pretty fragile to changes, and that it may look easy to redo something, but that may take a lot of time. and the developer may find out that the state that dev tried to change was one of the best options. I also have find-out I was wrong, and that the initial code went actually the right way.